src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Aerospike/AerospikeCommon.cs

```diff
@@ -0,0 +1,85 @@
+// <copyright file="AerospikeCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Tagging;
+using Datadog.Trace.Vendors.Serilog;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Aerospike
+{
+    internal class AerospikeCommon
+    {
+        private const string ServiceName = "aerospike";
+        private const string OperationName = "aerospike.command";
+        public const string IntegrationName = nameof(IntegrationIds.Aerospike);
+        internal static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(IntegrationName);
+
+        public static Scope CreateScope<TTarget>(Tracer tracer, out AerospikeTags tags, TTarget target)
+        {
+            tags = null;
+
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                tags = new AerospikeTags();
+                var serviceName = tracer.Settings.GetServiceName(tracer, ServiceName);
+                scope = tracer.StartActiveWithTags(OperationName, tags: tags, serviceName: serviceName);
+                var span = scope.Span;
+
+                if (target.TryGetFieldValue<object>("key", out var key) && key != null)
```

Comment:
Aren't we moving from this method to DuckTyping?

Reply:
Here I'm just testing if the "key" field is defined. It will be defined on some types and not on others. I find the ducktyping API a bit cumbersome for those scenarios 🤔

Reply:
🤔   we have the `TryDuckCast` method but fair enough your point.

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/1717/commits/5c7738a76002e45f9a89d3db6c793336ef3a6d17, but I wonder if we could build a helper to avoid having to define a new type when only interested in one single field.